packages/node-cli/src/CLI.js

```diff
@@ -37,9 +39,12 @@ module.exports = class CLI {
 
 	start(process) {
 		try {
-			const runInInteractiveMode = this._isRunningInInteractiveMode();
-
 			const commandMetadataList = this._commandsMetadataService.getCommandsMetadata();
+			this._validateCommandExists(commandMetadataList, process.argv[2]);
+			
+			process.argv = this._takeOutUnnecesaryArgumentsWhenHelpIsCalled();
```

Comment:
This line looks weird .... But I don't have a suggestion right now.

Reply:
I was naming it checkHelp... but I thought it would be better to be more explicit.

Reply:
Maybe we can rename it to retainHelpArgumentsOnly?
But I don't have a strong opinion on this.